src/conventions/h-group/clue-interpretation/connecting-cards.js

```diff
@@ -159,17 +160,82 @@ function find_unknown_connecting(game, giver, target, reacting, identity, connec
-			if (state.hands[giver].some(c => c.clued && game.players[giver].thoughts[c.order].inferred.has(identity))) {
+			// This may be a bluff. resolve_bluff will rule out a bluff if it doesn't
+			// look like it could lead to the following plays.
+			const bluff = game.level >= LEVEL.BLUFFS &&
+				connected.length == 1 &&
+				((giver + 1) % state.numPlayers) == reacting &&
+				state.hands[reacting].filter(c => game.players[reacting].unknown_plays.has(c.order) && game.players[reacting].thoughts[c.order].matches(identity)).length == 0;
+
+			// Could be duplicated in giver's hand - disallow hidden finesse unless it could be a bluff.
+			if (!bluff && state.hands[giver].some(c => c.clued && game.players[giver].thoughts[c.order].inferred.has(identity))) {
 				logger.warn(`disallowed hidden finesse on ${logCard(finesse)} ${finesse.order}, true ${logCard(identity)} could be duplicated in giver's hand`);
 				return;
 			}
 
-			return { type: 'finesse', reacting, card: finesse, hidden: true, identities: [finesse.raw()] };
+			return { type: 'finesse', reacting, card: finesse, hidden: true, bluff, identities: [finesse.raw()] };
 		}
 	}
 }
 
+/**
+ * Determines whether a bluff connection is a valid bluff, and updates the connection accordingly.
+ * @param {Game} game
+ * @param {number} giver 				The player index that gave the clue. They cannot deduce unknown information about their own hand.
+ * @param {number} target 				The player index receiving the clue. They will not find self-prompts or self-finesses.
+ * @param {Connection[]} connections	The complete connections leading to the play of a card.
+ * @param {ActualCard[]} promised		The non-hidden play(s) which are promised.
+ * @returns {Connection[]}
+ */
+export function resolve_bluff(game, giver, target, connections, promised) {
+	if (connections.length == 0 || !connections[0].bluff)
+		return connections;
+	const { state } = game;
+	const bluffCard = connections[0].card;
+	// Determine the next play if this is a bluff.
+	const next_play = connections.findIndex(connection => connection.card.order == promised[0].order) + 1;
+	// A bluff must be followed only by prompts.
+	let bluff_possible = connections.every((conn, index) => index < next_play || (!conn.hidden && conn.type !== 'finesse'));
+
+	if (bluff_possible) {
+		// A bluff must be recognizable. As such, there should be no connection
+		// between the bluffed card and at least one of the following plays
+		// as known by the player who would play next after the bluff play.
+		let known_bluff = false;
+		const next_player = state.hands.findIndex(hand => hand.find(c => c.order == promised[1].order));
+		const bluff_identies = (giver + 1) % state.numPlayers == state.ourPlayerIndex ?
+			game.players[state.ourPlayerIndex].thoughts[bluffCard.order].inferred.filter(c => state.isPlayable(c)) :
+			[bluffCard];
+		for (let i = 1; i < promised.length; ++i) {
+			const known_bluff_cards = bluff_identies.filter(identity => {
```

Comment:
Since we're not using `known_bluff_cards`, we can use `.some()` instead of `.filter()` here. Also, I think there's a spelling mistake in "bluff_**identities**".